### ISSUE TITLE
Deal with multi-position players cleanly

### ISSUE THREAD

**Issue by BenBrostoff** [BenBrostoff/draft-kings-fun#34]
Applies for NBA where DraftKings will list players as `PF/SF` or something similar - the optimizer should handle this by placing the player in the optimal position. Right now, it simply picks the first position (`PF` in the `PF/SF` example) and uses that one.

**Comment by rneu31** [comment]
Do you have any idea on how to solve this? I stumbled across this code after writing an almost equivalent (just for NBA) solver myself...

This post helps explain the math quite a bit .. might be helpful: https://orbythebeach.wordpress.com/2015/09/29/how-to-build-the-best-fantasy-football-team-part-2/

**Comment by BenBrostoff** [comment]
That's a fantastic post, although I must admit a lot of the math is over my head.

One super naive solution I've kicked around a bit is to create two instances of the player, one at each position, and then add an `is_multi_position` property to `Player`. I maybe then could use `ortools` to ensure it's impossible to end up with the same player at two different positions.

This is definitely a top priority for me as I waste a lot of time in NBA contests just overwriting CSV data and trying players at different positions 👎 . This will be a big problem again if I ever add a baseball option. If you have a better idea or want to take a crack at this, definitely all ears.

Also, interested to see if @swanson has any thoughts on this, as a lot of the code in this repo is based on his.

**Comment by swanson** [comment]
I think you're on a reasonable path Ben: I would have said make two "players" -- one for each position -- and then add an additional constraint that you can only have one of the two "duplicate" players. You could do that pretty easily in code I'd imagine.

**Comment by BenBrostoff** [comment]
@rneu31 I think [this PR I just merged resolves this issue](https://github.com/BenBrostoff/draft-kings-fun/pulls). Let me know if this works for you - I tested in a few NBA contests (which I'm playing tonight :-)) / wrote a unit test for it and seems to be okay.

@swanson Thanks for the guidance here, your recommendation worked great.

**Comment by rneu31** [comment]
@BenBrostoff I ended up writing a completely functioning equivalent using pulp because that's all I was familiar with. 

Figured I'd toss the design here since I shared the article that this was largely based on:
I created an array of bools for each possible position for each player, `Chris Paul: [x_ChrisPaul_PG, x_ChrisPaul_SG, ... x_ChrisPaul_C]`, detected the players actual position, in this case PG, and required that each of the invalid positions be zero, `x_ChrisPaul_SG = 0, ... x_ChrisPaul_C = 0` which ensures that Chris Paul can only be selected for the PG position. Then, I also ensured that `x_ChrisPaul_PG + ... + x_ChrisPaul_C <= 1` to ensure the player is not selected twice.

Will look into being an active contributor to this project instead of writing all of my own code (oops).

Thanks for your fix!
